src/datasets/utils/readme.py

```diff
@@ -0,0 +1,173 @@
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import yaml
+
+
+BASE_REF_URL = "https://github.com/huggingface/datasets/tree/master/src/datasets/utils"
+this_url = f"{BASE_REF_URL}/{__file__}"
+logger = logging.getLogger(__name__)
+
+
+def load_yaml_resource(resource: str) -> Tuple[Any, str]:
+    with open(resource) as f:
+        content = yaml.safe_load(f)
```

Comment:
You may need to use `pkg_resources` here to load the yaml data

See an example here:

https://github.com/huggingface/datasets/blob/8e903b5ef7c039cee79f0a2da3dd328d59c38588/src/datasets/utils/metadata.py#L25-L27

Reply:
Okay, I'll use `pkg_resources`, but can you please explain why it is needed?